skeleton/src/main/java/com/dropbox/kaiken/skeleton/skeleton/core/KaikenUserServicesProvider.kt

```diff
@@ -0,0 +1,67 @@
+package com.dropbox.kaiken.skeleton.skeleton.core
+
+import com.dropbox.kaiken.scoping.AppServices
+import com.dropbox.kaiken.scoping.TeardownHelper
+import com.dropbox.kaiken.scoping.UserServices
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.launch
+
+/**
+ * Skeleton implementation of [UserServicesProvider] that registers itself with a [UserManager].
+ *
+ */
+class KaikenUserServicesProvider
+constructor(
+        private val applicationServices: AppServices,
+        private val userServicesFactory: (AppServices, SkeletonUser) -> UserServices,
+) : SkeletonUserServicesProvider {
+    private val userServicesMap = mutableMapOf<String, KaikenUserServices>()
+
+    init {
+        //TODO Mike figure out if we want another scope to launch from
+        GlobalScope.launch {
+            (applicationServices as KaikenAppServices).userManager().getUserState().collect { userState ->
+                userState.usersRemoved.forEach { user ->
```

Comment:
What do you think about moving the synchronized blocks here instead of in each method? I'd assume we'd want the entire set of add / remove to be atomic.

Reply:
done

Reply:
can't synchronize inside of a scope apparently, strangely lets me do it inside function that are called.